lib/Optimizer/Transforms/GenKernelExecution.cpp

```diff
@@ -1207,8 +1217,9 @@ class GenerateKernelExecution
         hasTrailingData = true;
         continue;
       }
-      if (isa<cudaq::cc::PointerType>(inTy))
+      if (isa<cudaq::cc::PointerType>(inTy) && !isStatePointerType(inTy))
```

Comment:
This doesn't look correct. We cannot pass a `cudaq::state*` as a pointer-free value.